docker/rust_dockerfile

```diff
@@ -39,14 +41,14 @@ RUN go get golang.org/x/tools/go/gcimporter15 \
 
 RUN curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path --default-toolchain nightly-2018-01-12 -y \
 	&& /root/.cargo/bin/rustup default nightly-2018-01-12 \
-	&& /root/.cargo/bin/cargo install --vers 0.3.4 --force rustfmt-nightly
+	&& /root/.cargo/bin/rustup component add rustfmt-preview
```

Comment:
we forcibly use nightly 0.3.4 version for rustfmt here

Reply:
Do you want me to remove this commit from my PR or do you want to make a new PR with cherry-picks from this branch?

Reply:
I think we should keep the origin, still use the 0.3.4.